src/ddmd/traits.d

```diff
@@ -403,6 +403,33 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
         return new IntegerExp(e.loc, false, Type.tbool);
     }
 
+    TypeFunction firstArgFunctionType()
```

Comment:
> do you agree ?

Mostly. The diff didn't show that there's also support for function symbols in the other traits.

You should remove the handling for FuncDeclaration in traits getLinkage, getFunctionVariadicStyle and getParameterStorageClasses, though. It is duplicated now.

> the reason I did not do this is because I'd have to import rootobject

That's not really a good excuse. You can also use typeof(args[0]) to avoid an import... ;-)